lib/libc/minimal/source/time/asctime.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2021 Sun Amar
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <errno.h>
+#include <time.h>
+#include <bits/restrict.h>
+
+static char *z_impl_asctime(const struct tm *timeptr, char *buf, size_t buflen)
+{
+	static char wday_name[7][3] = {
+		"Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"
+	};
+	static char mon_name[12][3] = {
+		"Jan", "Feb", "Mar", "Apr", "May", "Jun",
+		"Jul", "Aug", "Sep", "Oct", "Nov", "Dec"
+	};
+
+	if (timeptr == NULL) {
+		errno = EINVAL;
+		return NULL;
+	}
```

Comment:
Also at minimum validate that all the values you're using as indexes have legal ranges.  You can either reject in that case, or emit `???`, but do not index an invalid pointer.  GNU libc doesn't seem to validate other fields.